File: MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/BaseModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace gov.va.medora.mdo.domain.sm
{
    public class BaseModel : PersistentObject
    {
        public bool Active { get; set; }
        private DateTime CreatedDate { get; set; }
        private DateTime ModifiedDate { get; set; }

        public BaseModel()
        {
            Active = true;
            // TBD - this seems wrong... from MHV code... Will new created date get set every time?? 
            CreatedDate = new DateTime();
            ModifiedDate = new DateTime();
        }
    }
}

File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/trash/BaseModel.java
package gov.va.med.mhv.sm.model;

import java.io.Serializable;
import java.util.Date;

import gov.va.med.mhv.foundation.model.PersistentObject;


/**
 * 
 * @author vhalommccarw
 *
 * Establishes a base from which every table will have
 * an active, dateCreated, and dateModified field and
 * it will be updated automatically.
 *
 */

public abstract class BaseModel extends PersistentObject implements Serializable {
	
	private boolean active;
	private Date createdDate;
	private Date modifiedDate;
	
	public BaseModel(){
		this.active = true;
		createdDate = new Date();
		modifiedDate = new Date();
	}
	
	public boolean isActive() {
		return active;
	}
	
	public void setActive(boolean active) {
		this.active = active;
	}
	
	public Date getCreatedDate() {
		return createdDate;
	}
	
	public void setCreatedDate(Date dateCreated) {
		this.createdDate = dateCreated;
	}
	
	public Date getModifiedDate() {
		return modifiedDate;
	}
	
	public void setModifiedDate(Date dateModified) {
		this.modifiedDate = dateModified;
	}	
}
